airflow/config_templates/default_airflow.cfg

```diff
@@ -667,6 +667,23 @@ statsd_allow_list =
 # def func_name(stat_name: str) -> str:
 stat_name_handler =
 
+# Dogstatsd
+# (https://docs.datadoghq.com/developers/metrics/dogstatsd_metrics_submission/?tab=python#metric-tagging)
+dogstatsd_on = False
+dogstatsd_host = localhost
```

Comment:
Why are these separate settings to the normal statsd/stats options?

Reply:
This PR is not replacing the statsd, but adding the support of dogstatsd along with statsd. So, it will work as it is for existing users who are using statsd. Also, the idea is to provide a choice to use anyone based on this configs without affecting existing users who are using statsd.

Reply:
I would instead have done this as a single flag -- such as `stasts_datadog_enabled = True`, because all the rest of the configs are the same, and I can't see a case when a user would want both plain statsd and dogstatsd enable.

Thoughts?

Reply:
Yes, there are some common configs and no need to duplicate it. I like your idea to add necessary configs only.
I am okay with your suggestions. If we go with this,  we just need only two more configs. 

`
stasts_datadog_enabled = False
  stasts_datadog_tags =
`

Reply:
What do you think about these configs @ashb ?

Reply:
@ashb I have made changes to use only these two variables. 
`stasts_datadog_enabled = False`
`stasts_datadog_tags =`